src/Controls/src/Core/HybridWebView/HybridWebView.cs

```diff
@@ -15,40 +25,215 @@ public class HybridWebView : View, IHybridWebView
+
+			await EvaluateJavaScriptAsync($"window.HybridWebView.InvokeMethod({currentInvokeTaskId}, {methodName}, [{paramsValuesStringArray}])");
+
+			return await callback.Task;
+		}
+#else
+		public Task<string?> InvokeJavaScriptAsync(string methodName, object?[]? paramValues = null, object?[]? paramJsonTypeInfos = null)
+		{
+			_invokeTaskId++; // This is to avoid the compiler warning about the field not being used
+			throw new NotImplementedException();
+		}
+#endif
+
+		/// <summary>
+		/// Invokes a JavaScript method named <paramref name="methodName"/> and optionally passes in the parameter values specified
+		/// by <paramref name="paramValues"/> by JSON-encoding each one.
+		/// </summary>
+		/// <typeparam name="TReturnType">The type of the return value to deserialize from JSON.</typeparam>
+		/// <param name="methodName">The name of the JavaScript method to invoke.</param>
+		/// <param name="returnTypeJsonTypeInfo">Metadata about deserializing the type of the return value specified by <typeparamref name="TReturnType"/>.</param>
+		/// <param name="paramValues">Optional array of objects to be passed to the JavaScript method by JSON-encoding each one.</param>
+		/// <param name="paramJsonTypeInfos">Optional array of metadata about serializing the types of the parameters specified by <paramref name="paramValues"/>.</param>
+		/// <returns>An object of type <typeparamref name="TReturnType"/> containing the return value of the called method.</returns>
+#if WINDOWS || ANDROID || IOS || MACCATALYST
+		public async Task<TReturnType?> InvokeJavaScriptAsync<TReturnType>(string methodName, JsonTypeInfo<TReturnType?> returnTypeJsonTypeInfo, object?[]? paramValues = null, JsonTypeInfo?[]? paramJsonTypeInfos = null)
+		{
+			var stringResult = await InvokeJavaScriptAsync(methodName, paramValues, paramJsonTypeInfos);
+
+			if (stringResult is null)
+			{
+				return default;
+			}
+			return JsonSerializer.Deserialize<TReturnType?>(stringResult, returnTypeJsonTypeInfo);
+	}
+#else
+		public Task<TReturnType?> InvokeJavaScriptAsync<TReturnType>(string methodName, object returnTypeJsonTypeInfo, object?[]? paramValues, object?[]? paramJsonTypeInfos)
+		{
+			throw new NotImplementedException();
+		}
+#endif
+
+		/// <inheritdoc/>
+		public async Task<string?> EvaluateJavaScriptAsync(string script)
+		{
+			if (script == null)
+			{
+				return null;
+			}
+
+			// Make all the platforms mimic Android's implementation, which is by far the most complete.
+			if (DeviceInfo.Platform != DevicePlatform.Android)
+			{
+				script = WebView.EscapeJsString(script);
+
+				if (DeviceInfo.Platform != DevicePlatform.WinUI)
+				{
+					// Use JSON.stringify() method to converts a JavaScript value to a JSON string
+					script = "try{JSON.stringify(eval('" + script + "'))}catch(e){'null'};";
+				}
+				else
+				{
+					script = "try{eval('" + script + "')}catch(e){'null'};";
+				}
+			}
```

Comment:
Most of this function, and especially this code should be in Core via a command. The handler should be doing this instead of checking the platform. Android knows it does not need this, so it does not need to have this code.